packages/outputarea/src/widget.ts

```diff
@@ -471,11 +471,11 @@ namespace OutputArea {
     };
 
     if (!session.kernel) {
-      return Promise.resolve(void 0);
+      return Promise.reject('Session has no kernel.');
```

Comment:
I felt this was better than forcing the return promise to be `Promise<Msg | undefined>`